bugsnag-android-unity/src/main/java/com/bugsnag/android/unity/UnityClient.java

```diff
@@ -10,32 +10,119 @@
 
 public class UnityClient {
 
+    static volatile Client client;
+    private static final Object lock = new Object();
+
     public static void init(Context androidContext, String apiKey, boolean trackSessions) {
-        Configuration config = new Configuration(apiKey);
-        config.setAutoCaptureSessions(trackSessions);
-        Bugsnag.init(androidContext, config);
+        if (client == null) {
+            synchronized (lock) {
+                if (client == null) {
+                    Configuration config = new Configuration(apiKey);
+                    config.setAutoCaptureSessions(trackSessions);
+                    client = new Client(androidContext, config);
```

Comment:
How come this is using `new Client` rather than `Bugsnag.init` (which returns a `Client`)?

Reply:
good catch! Looks like it changed here https://github.com/bugsnag/bugsnag-unity/pull/73/commits/27e32c4939598ef668fc95781f841aec352e8914#r185431610, @kattrali do you know the reasoning?